plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt

```diff
@@ -421,13 +419,9 @@ class FeatureDevController(
                 credentialStartUrl = getStartUrl(project = context.project)
             )
 
-            // Caution: insertChanges has multiple responsibilities.
-            // The filter here results in rejected files being hidden from the tree after continuing, by design.
-            // However, it is critical that we don't hide already-accepted files. Inside insertChanges, it
-            // filters to only update the subset of passed files that aren't accepted or rejected already.
-            session.insertChanges(
-                filePaths = filePaths.filter { !it.rejected },
-                deletedFiles = deletedFiles.filter { !it.rejected },
+            session.insertChangesAndUpdateFileComponents(
```

Comment:
nit: I recognize that we're trying to maintain the existing logic of both writing to disk and updating the file component because it appears to be appropriate here. Is it cleaner to avoid introducing the "And" method and just perform the individual operations here? 

If we do introduce this method, we add a choice to future authors to either update separately or together and I don't think we've sufficiently explained when it should be performed together versus not. 

```
session.writeChangesToDisk(...)
val codeResultMessageId = this._codeResultMessageId
if (codeResultMessageId != null) {
    messenger.updateFileComponent(...)
}
```